axelrod/fingerprint.py

```diff
@@ -14,219 +16,355 @@
 Point = namedtuple('Point', 'x y')
 
 
-def create_points(step, progress_bar=True):
-    """Creates a set of Points over the unit square.
+PlayerOrStrategy = Union[Player, type]
+Edges = Tuple[int, int]
+Matches = List[List[Tuple[Action, Action]]]
+
+PointsPlayers = Dict[Point, Player]
+PointsFloats = Dict[Point, float]
+PointsEdges = Dict[Point, Edges]
+PointsMatches = Dict[Point, Matches]
+
+
+class AshlockFingerprint(object):
+    def __init__(self, strategy: PlayerOrStrategy,
+                 probe: PlayerOrStrategy) -> None:
+        self._strategy = strategy
+        self._probe = probe
+        self._data = None  # type: DataOrganizer
+
+    @property
+    def interactions(self) -> PointsMatches:
+        if not self._data:
+            interactions = None  # type: PointsMatches
+            return interactions
+        return self._data.get_points_interactions_dict()
+
+    @property
+    def data(self) -> PointsFloats:
+        if not self._data:
+            data = None  # type: PointsFloats
+            return data
+        return self._data.get_points_averages_dict()
+
+    def fingerprint(self, turns: int = 50, repetitions: int = 10,
+                    step: float = 0.01, processes: int = None,
+                    filename: str = None, in_memory: bool = False,
+                    progress_bar: bool = True) -> PointsFloats:
+
+        tournament_creator = SpatialTournamentCreator(self._strategy,
+                                                      self._probe,
+                                                      step)
+        tournament_kwargs = {'turns': turns, 'repetitions': repetitions}
+
+        play_kwargs = _get_play_kwargs(filename=filename, in_memory=in_memory,
```

Comment:
i don't like this, but i don't know why.  it feels clunky.  i'm not sure if anything is actually bad here, but would appreciate some input.